test/json_unittest.cc

```diff
@@ -0,0 +1,385 @@
+  try {
+    test();
+  } catch (const json::Exception& exc) {
+    FAIL() << exc.what();
+  }
+}
+
+
+// Trival Null
+
+TEST(TrivialParseTest, Null) {
+  GuardJsonException([](){
+    std::unique_ptr<json::Value> v = json::Parser::FromString("null");
+    EXPECT_EQ(v->type(), json::NullType);
+  });
+}
+
+TEST(TrivialToStringTest, Null) {
+  GuardJsonException([](){
+    EXPECT_TO_STRING(json::null(), "null");
+  });
+}
+
+
+// Trival True & False
+
+TEST(TrivialParseTest, True) {
+  GuardJsonException([](){
+    std::unique_ptr<json::Value> v = json::Parser::FromString("true");
+    EXPECT_EQ(v->As<json::Boolean>()->value(), true);
+  });
+}
+
+TEST(TrivialToStringTest, True) {
+  GuardJsonException([](){
+    EXPECT_TO_STRING(json::boolean(true), "true");
+  });
+}
+
+TEST(TrivialParseTest, False) {
+  GuardJsonException([](){
+    std::unique_ptr<json::Value> v = json::Parser::FromString("false");
+    EXPECT_EQ(v->As<json::Boolean>()->value(), false);
+  });
+}
+
+TEST(TrivialToStringTest, False) {
+  GuardJsonException([](){
+    EXPECT_TO_STRING(json::boolean(false), "false");
+  });
+}
+
+
+// Trival Number
+
+TEST(TrivialParseTest, Number) {
+  GuardJsonException([](){
+    std::unique_ptr<json::Value> v = json::Parser::FromString("2");
+    EXPECT_EQ(v->As<json::Number>()->value(), 2.0);
+  });
+}
+
+TEST(TrivialToStringTest, Number) {
+  GuardJsonException([](){
+    EXPECT_TO_STRING(json::number(2.0), "2.0");
+  });
+}
+
+
+// Trival String
+
+TEST(TrivialParseTest, StringEmpty) {
+  GuardJsonException([](){
+    std::unique_ptr<json::Value> v = json::Parser::FromString("\"\"");
+    EXPECT_EQ(v->As<json::String>()->value().length(), 0);
+  });
+}
+
+TEST(TrivialToStringTest, StringEmpty) {
+  GuardJsonException([](){
+    EXPECT_TO_STRING(json::string(""), "\"\"");
+  });
+}
+
+TEST(TrivialParseTest, StringOneChar) {
+  GuardJsonException([](){
+    std::unique_ptr<json::Value> v = json::Parser::FromString("\"x\"");
+    const auto& str = v->As<json::String>();
+    EXPECT_EQ(str->value(), "x");
+  });
+}
+
+TEST(TrivialToStringTest, StringOneChar) {
+  GuardJsonException([](){
+    EXPECT_TO_STRING(json::string("x"), "\"x\"");
+  });
+}
+
+
+// Trival Array
+
+TEST(TrivialParseTest, ArrayEmpty) {
+  GuardJsonException([](){
+    std::unique_ptr<json::Value> v = json::Parser::FromString("[]");
+    EXPECT_TRUE(v->As<json::Array>()->empty());
+  });
+}
+
+TEST(TrivialToStringTest, ArrayEmpty) {
+  GuardJsonException([](){
+    EXPECT_TO_STRING(json::array({}), "[]");
+  });
+}
+
+TEST(TrivialParseTest, ArrayOneElement) {
+  GuardJsonException([](){
+    std::unique_ptr<json::Value> v = json::Parser::FromString("[2]");
+    const auto& arr = v->As<json::Array>();
+    EXPECT_EQ(arr->size(), 1);
+    EXPECT_EQ((*arr)[0]->As<json::Number>()->value(), 2.0);
+  });
+}
+
+TEST(TrivialToStringTest, ArrayOneElement) {
+  GuardJsonException([](){
+    EXPECT_TO_STRING(json::array({json::number(2.0)}), "[2.0]");
+  });
+}
+
+
+// Trival Dict
```

Comment:
Nit: "Dict" is Python. In JSON, these are "Object"s.

Reply:
Done.